src/Api/BrandsApi.php

```diff
@@ -0,0 +1,350 @@
+    {
+        return Template::fromArray($this->client->get("/v2/customers/{$customer}/brands/{$brand}/templates/{$name}")->json());
+    }
+
+    /**
+     * @see https://dm.realtimeregister.com/docs/api/brands/templates/list
+     *
+     * @param string      $customer
+     * @param string      $brand
+     * @param int|null    $limit
+     * @param int|null    $offset
+     * @param string|null $search
+     *
+     * @return TemplateCollection
+     */
+    public function listTemplates(
+        string $customer,
+        string $brand,
+        ?int $limit = null,
+        ?int $offset = null,
+        ?string $search = null
+    ): TemplateCollection {
+        $query = [];
+        if (! is_null($limit)) {
+            $query['limit'] = $limit;
+        }
+        if (! is_null($offset)) {
+            $query['offset'] = $offset;
+        }
+        if (! is_null($search)) {
+            $query['q'] = $search;
+        }
+
+        $response = $this->client->get("/v2/customers/{$customer}/brands/{$brand}/templates", $query);
+        return TemplateCollection::fromArray($response->json());
+    }
+
+    /**
+     * @see https://dm.realtimeregister.com/docs/api/brands/templates/update
+     *
+     * @param string      $customer
+     * @param string      $brand
+     * @param string      $name
+     * @param string|null $subject
+     * @param string|null $text
+     * @param string|null $html
+     */
+    public function updateTemplate(
+        string $customer,
+        string $brand,
+        string $name,
+        ?string $subject,
+        ?string $text,
+        ?string $html
+    ): void {
+        $payload = [];
+
+        if ($subject) {
+            $payload['subject'] = $subject;
+        }
+        if ($text) {
+            $payload['text'] = $text;
+        }
+        if ($html) {
+            $payload['html'] = $html;
+        }
+
+        $this->client->post("/v2/customers/{$customer}/brands/{$brand}/templates/{$name}/update", $payload);
+    }
+
+    /**
+     * @see https://dm.realtimeregister.com/docs/api/brands/templates/preview
+     *
+     * @param string      $customer
+     * @param string      $brand
+     * @param string      $name
+     * @param string|null $contexts
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return TemplatePreview
+     */
+    public function previewTemplate(string $customer, string $brand, string $name, ?string $contexts = null): TemplatePreview
+    {
+        $invalidTemplateNamesForPreview = [
+            TemplateNameEnum::TEMPLATE_NAME_EMAIL_HEADER,
+            TemplateNameEnum::TEMPLATE_NAME_EMAIL_FOOTER,
+            TemplateNameEnum::TEMPLATE_NAME_WEB_HEADER,
+            TemplateNameEnum::TEMPLATE_NAME_WEB_FOOTER,
+        ];
+        if (in_array($name, $invalidTemplateNamesForPreview)) {
+            throw new InvalidArgumentException('Template not available for preview.');
```

Comment:
Could you also mention the given template name and the forbidden values. Kind of like we do here: https://github.com/sandwave-io/realtimeregister-php/blob/master/src/Domain/Enum/AbstractEnum.php#L19

Reply:
Added: https://github.com/sandwave-io/realtimeregister-php/pull/25/commits/1655a478e41745d8747824e8e14c5f61d505ea81